boot_node/src/config.rs

```diff
@@ -58,19 +57,16 @@ impl<T: EthSpec> TryFrom<&ArgMatches<'_>> for BootNodeConfig<T> {
         let logger = slog_scope::logger();
 
         set_network_config(&mut network_config, matches, &data_dir, &logger, true)?;
-        // default to the standard port
+
+        // Set the enr-udp-port to the default listening port if it was not specified.
         if !matches.is_present("enr-udp-port") {
-            network_config.enr_udp_port = Some(
-                matches
-                    .value_of("port")
-                    .expect("Value required")
-                    .parse()
-                    .map_err(|_| "Invalid port number")?,
-            );
+            network_config.enr_udp_port = Some(network_config.discovery_port);
         }
 
-        let auto_update = matches.is_present("enable-enr_auto_update");
-        let disable_packet_filter = matches.is_present("disable-packet-filter");
+        // By default this is enabled. If it is not set, revert to false.
+        if !matches.is_present("enable-enr_auto_update") {
+            network_config.discv5_config.enable_packet_filter = false;
+        }
```

Comment:
```suggestion
        if !matches.is_present("enable-enr-auto-update") {
            network_config.discv5_config.enr_update = false;
        }
```

I think this flag name is wrong, and it disables the packet filter instead of the ENR auto update.

I might try to add some CLI regression tests for the boot node to [`lighthouse/tests`](https://github.com/sigp/lighthouse/tree/stable/lighthouse/tests), because it seems this flag was already broken before this change.

I agree that we don't need to process the `--disable-packet-filter` flag because [`set_network_config`](https://github.com/sigp/lighthouse/blob/f9bba92db3468321b28ddd9010e26b359f88bafe/beacon_node/src/config.rs#L619-L622) takes care of it.

Reply:
Nice catch! Thanks

Reply:
Opened this issue to track the testing https://github.com/sigp/lighthouse/issues/2602